src/AccessibilityInsights.SharedUx/Telemetry/TelemetryController.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+using System;
+
+namespace AccessibilityInsights.SharedUx.Telemetry
+{
+    public static class TelemetryController
+    {
+        public static void EnableTelemetry()
+        {
+            // Open the telemetry sink
+            TelemetrySink.IsTelemetryAllowed = true;
+
+            // Begin listening for telemetry events
+            // This must be done after the low-level sink is opened above
+            // So that queued events get flushed to an open telemetry sink
+            EventTelemetrySink.Enable();
+
+            AxeWindowsTelemetrySink.Enable();
+        }
+
+        public static void DisableTelemetry()
+        {
+            // Close the telemetry sink
+            TelemetrySink.IsTelemetryAllowed = false;
```

Comment:
>IsTelemetryAllowed [](start = 26, length = 18)

Consider the scenario where someone starts with telemetry enabled, then disables it, then enables it again. We end up with unbalanced calls to EventTelemetrySink.Enable and AxeWindowsTelemetrySink.Enable. Maybe that's OK, maybe not. In order to minimize assumptions between these classes, I'd suggest having and calling EventTelemetrySink.Disable() and AxeWindowsTelemetrySink.Disable()

Reply:
I'd like to point out that this was a problem before the refactor. The refactor just made it visible. As the code currently stands, the on-off-on scenario won't result in excess telemetry.

Reply:
Fair enough

---
In reply to: [272386735](https://github.com/Microsoft/accessibility-insights-windows/pull/258#discussion_r272386735) [](ancestors = 272386735)